pkgs/by-name/libresoc/ieee754fpu.nix

```diff
@@ -0,0 +1,40 @@
+{
+  python39Packages,
+  fetchgit,
+  symbiyosys,
+  nmutil,
+  nmigen,
+  pytest-output-to-files,
+}:
+with python39Packages;
+  buildPythonPackage rec {
+    name = "libresoc-ieee754fpu";
+    pname = "ieee754fpu";
+    version = "unstable-2024-03-31";
+
+    # given (as of ${version} date) inconsistent upstream reliability
+    # at https://git.libre-soc.org/git/${pname}.git,
+    # Libre-SOC repos mirrored to github.com/Libre-SOC-mirrors
+    # to ease switching between upstream and mirror,
+    # use the compatible fetchgit rather than fetchFromGitHub
+    src = fetchgit {
+      url = "https://github.com/Libre-SOC-mirrors/${pname}.git";
+      hash = "sha256-Ghbvg2Y4YlmxVEa3EtcvEVai4hC4VU4q+XIQh4pQ7+c=";
+      rev = "829dfbc53ba38ec17bc544cb0b862e73cee223db"; # HEAD @ version date
+    };
+
+    prePatch = ''
+      touch ./src/ieee754/part{,_ass,_cat,_repl}/__init__.py
+    '';
+
+    propagatedBuildInputs = [nmutil];
+
+    nativeCheckInputs = [pytestCheckHook pytest-xdist pytest-output-to-files nmigen symbiyosys];
+
+    # TODO(jleightcap): all tests pass except formal methods,
```

Comment:
Why not run all tests except those, then? :)

Reply:
`sfpy` and `bigfloat` look good to me. Nice improvement to only disable the failing upstream tests!